wip-index.d.ts

```diff
@@ -13,33 +13,250 @@ export = DailyIframe;
+    properties?: DailyIframe.FrameProps
+  ): DailyIframe;
+  static createFrame(properties?: DailyIframe.FrameProps): DailyIframe;
+  static createTransparentFrame(
+    properties?: DailyIframe.FrameProps
+  ): DailyIframe;
+
+  // Instance methods
+  iframe(): HTMLIFrameElement | null;
   join(
     properties?: DailyIframe.FrameProps
   ): Promise<DailyIframe.Participant[] | void>;
   leave(): Promise<void>;
   destroy(): Promise<void>;
   meetingState(): DailyIframe.MeetingState;
-  participants(): { [id: string]: DailyIframe.Participant };
+  participants(): {
+    local: DailyIframe.Participant;
+    [id: string]: DailyIframe.Participant;
+  };
+  updateParticipant(
+    sessionId: string,
+    updates: DailyIframe.ParticipantUpdates
+  ): DailyIframe;
+  updateParticipants(updates: {
+    [sessionId: string]: DailyIframe.ParticipantUpdates;
+  }): DailyIframe;
+  localAudio(): boolean;
+  localVideo(): boolean;
   setLocalAudio(enabled: boolean): DailyIframe;
   setLocalVideo(enabled: boolean): DailyIframe;
-  startCamera(properties?: DailyIframe.FrameProps): Promise<any>; // TODO: flesh out return type
-  on(event: DailyIframe.Event, handler: (event?: any) => void): DailyIframe; // TODO: flesh out handler
-  once(event: DailyIframe.Event, handler: (event?: any) => void): DailyIframe; // TODO: flesh out handler
-  off(event: DailyIframe.Event, handler: (event?: any) => void): DailyIframe; // TODO: flesh out handler
+  setBandwidth(bw: {
+    kbs?: number | "NO_CAP" | null;
+    trackConstraints?: MediaTrackConstraints;
+  }): DailyIframe;
+  setDailyLang(lang: DailyIframe.Language): DailyIframe;
+  startCamera(
+    properties?: DailyIframe.FrameProps
+  ): Promise<DailyIframe.DeviceInfos>;
+  cycleCamera(): Promise<{ device?: MediaDeviceInfo | null }>;
+  cycleMic(): Promise<{ device?: MediaDeviceInfo | null }>;
+  setInputDevices(devices: {
+    audioDeviceId?: string;
+    audioSource?: MediaStreamTrack;
+    videoDeviceId?: string;
+    videoSource?: MediaStreamTrack;
+  }): DailyIframe;
+  setOutputDevice(audioDevice: { id?: string }): DailyIframe;
+  getInputDevices(): Promise<DailyIframe.DeviceInfos>;
+  load(properties: DailyIframe.FrameProps): Promise<void>;
+  startScreenShare(captureOptions?: DailyIframe.ScreenCaptureOptions): void;
+  stopScreenShare(): void;
+  startRecording(): void;
+  stopRecording(): void;
+  getNetworkStats(): Promise<DailyIframe.NetworkStats>;
+  getActiveSpeaker(): { peerId?: string };
+  setActiveSpeakerMode(enabled: boolean): DailyIframe;
+  activeSpeakerMode(): boolean;
+  subscribeToTracksAutomatically(): boolean;
+  setSubscribeToTracksAutomatically(enabled: boolean): DailyIframe;
+  enumerateDevices(): Promise<{ devices: MediaDeviceInfo[] }>;
+  sendAppMessage(data: any, to?: string): DailyIframe;
+  addFakeParticipant(details?: { aspectRatio: number }): DailyIframe;
+  setShowNamesMode(mode: false | "always" | "never"): DailyIframe;
+  detectAllFaces(): Promise<{
```

Comment:
wait. what?

Reply:
I know, right? This typing exercise was a fun way to dig into every nook and cranny of our API, which is a Swiss army knife it turns out!

Reply:
For a second I was wondering whether to expose some "hidden" (undocumented) functionality through the type declaration, but then figured I'd go ahead and do it since `daily-js` is open-source so nothing is "truly" hidden anyway.